### ISSUE TITLE
The volume renderer doesn't allow non-black background colors.

### ISSUE THREAD

**Issue by atmyers** [yt-project/yt#1579]
This used to work, but seems to have been broken. See the docs page here for scripts and example outputs: http://yt-project.org/doc/cookbook/simple_plots.html#image-background-colors

**Comment by atmyers** [comment]
This is related to the fact that the images used to have alpha values of 0 in the background pixels when returning from the volume renderer, but they now have 1s everywhere. Has there been a recent change that could have affected this?

**Comment by ngoldbaum** [comment]
The easiest way to track that down is probably to bisect.

Just a guess, maybe the fixes I did to make compositing different VolumeSource instances together?

**Comment by ngoldbaum** [comment]
It looks like this was broken as far back as yt 3.3.1, which means it broke sometime during the volume rendering refactor. This makes it kind of painful to bisect, unfortunately.

**Comment by ngoldbaum** [comment]
I was able to get this to kinda work, but only for renderings of a VolumeSource with a transfer function that has `grey_opacity=True`. Unfortunately the old API (that applied the background colors to the image after doing the rendering) won't work with the compositing volume renderer (I think anyway), so it requires a change to the API to set the background color before doing the rendering. I ended up implementing it as a property that hangs off the scene object.

I made the following modifications: https://gist.github.com/ngoldbaum/3de5775267483186af4c77a773e0b7a8

Unfortunately this is unsatisfying in that it only seems to work properly with renderings that have `grey_opacity = True`. Non-grey renderings end up with a uniform image filled with the background color. Here's a sample script that works with the above patch:

```
import yt

ds = yt.load('Enzo_64/DD0043/data0043')

sc = yt.create_scene(ds, lens_type='perspective')

# Get a reference to the VolumeSource associated with this scene
# It is the first source associated with the scene, so we can refer to it
# using index 0.
source = sc[0]

# Set the bounds of the transfer function
source.tfh.set_bounds((3e-31, 5e-27))

# set that the transfer function should be evaluated in log space
source.tfh.set_log(True)

# Make underdense regions appear opaque
source.tfh.grey_opacity = True

sc.background_color = 'red'

# save the image, flooring especially bright pixels for better contrast
sc.save('rendering.png', sigma_clip=6.0)
```

![rendering](https://user-images.githubusercontent.com/3126246/34062564-cf027900-e1b2-11e7-9f3d-d8ee0ab7c3dd.png)

It would be nice if we could figure out a solution that works for all kinds of renderings. It would be even nicer if there was a way to do it without changing the API, although I fear that won't be possible.

**Comment by aparamon** [comment]
Currently, it is impossible to specify white background. As a workaround, I managed to get the desired result by inverting the rendered picture (post-processing). However, adjusting transfer function colors in inverse mode proved extremely hard.

Ability to specify white background for both 2D and 3D plots consistently is of crucial importance for publication-quality graphics.

**Comment by muuud** [comment]
Sharing as per ngoldbaum's request from Slack. I encountered this issue and was able to find a different workaround using line objects. The script I used is attached here [https://gist.github.com/muuud/5b4fb726351599395b51ae6925ab1ce5](url)

The nature of the workaround involved drawing three thousand line objects with the appropriate colors and opacities in particular locations in the render image, to spoof the appearance of a background color.

![download-2](https://user-images.githubusercontent.com/33235250/39075530-06f5f3b8-44ab-11e8-86da-d9804ffd318e.png)

**Comment by emotheau** [comment]
Hi, I'm also facing the same issue. I want to have a white background to create a publishable picture of my simulated 3D flame. Are you planning to fix this problem soon or do I need to use other tools instead?
Thanks !

**Comment by ngoldbaum** [comment]
There’s no immediate plan to fix this, although I’m hoping the example @muuud shared will give me some ideas.

**Comment by ngoldbaum** [comment]
I didn't get a chance to take a closer look. Since this is not really a bug per se I'm going to punt on doing this for 3.5.

**Comment by evanmayer** [comment]
I would also like this feature for making publication-ready plots with a white background. I would consider it a bug, especially because as per the OP, the documentation that purports to demonstrate the feature does not.

**Comment by matthewturk** [comment]
@evanmayer I agree, and I see that @neutrinoceros has changed the label.  I'll see if I can take a look at this right away.

**Comment by matthewturk** [comment]
I made a few changes to the code to experiment with possible solutions; what I'm finding (which is likely not news to others who have experimented with this) is that the compositing doesn't come across strongly enough when we're compositing it on any color other than black.

That being said, starting with the transparent one, I was able to get some nice images by using the GNU Image Manipulator and playing with how it compositing onto background layers.

![transparent_bg](https://github.com/yt-project/yt/assets/89019/f528a31d-2094-428f-bf81-564d48b3fd72)
![white_bg](https://github.com/yt-project/yt/assets/89019/50048c4e-bf60-4921-8498-502b94e4955d)

I'll try to work on this again soon.

**Comment by neutrinoceros** [comment]
@Socob rightfully pointed out that #3851 was a duplicate of this issue. See in particular [this comment(https://github.com/yt-project/yt/issues/3851#issuecomment-1164911201) from @chrishavlin which contain some valuable insight.
